raven/src/main/java/com/getsentry/raven/RavenFactory.java

```diff
@@ -109,6 +109,20 @@ public static Raven ravenInstance(Dsn dsn, String ravenFactoryName) {
             }
         }
 
+        if (ravenFactoryName != null && triedFactories.size() < 1) {
+            try {
+                // see if the provided class exists on the classpath at all
+                Class.forName(ravenFactoryName);
+                logger.error(
+                    "The raven factory class '{}' is missing a ServiceLoader provider "
+                    + "for com.getsentry.raven.RavenFactory. "
+                    + "See: https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html", ravenFactoryName);
```

Comment:
Forgot about manual. Going to link to our own docs and explain both solutions. Did the other stuff too.